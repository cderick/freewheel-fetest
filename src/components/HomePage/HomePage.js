import React from 'react'
import PropTypes from 'prop-types'
import Navigation from './Navigation/Navigation'
import TFLBanner from './TFLBanner/TFLBanner'
import HeroBanner from './HireBike/HeroBanner/HeroBanner'
import DataBanner from './HireBike/DataBanner/DataBanner'
import data from '../../constants/data'

class HomePage extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			heroBanner: undefined,
			navItems: undefined,
			tubeBanner: undefined,
			defaultNav: 'tflInfo'
		}
		this.changeNav = this.changeNav.bind(this)
	}

	componentDidMount() {
		const { mainNav, heroBanner, tubeBanner } = data
		this.setState({
			heroBanner: heroBanner,
			navItems: mainNav,
			tubeBanner: tubeBanner
		})
	}

	changeNav(nav) {
		const { defaultNav } = this.state
		if(defaultNav !== nav){
			this.setState({
				defaultNav: nav,
			})
		}
	}

	render() {
		const { defaultNav, navItems, tubeBanner, heroBanner } = this.state
		const { 
			tflInfo, 
			cleanData, 
			getApiQuery, 
			showContent, 
			queryValue, 
			retrievedObjet 
		} = this.props
		return (
			<section>
				<Navigation changeNav={this.changeNav} navItems={navItems} />
				{defaultNav === 'tflInfo' ? (
					<section>
						<TFLBanner tflInfo={tflInfo} tubeBanner={tubeBanner} />
					</section>
				) : (
					<section>
						<HeroBanner cleanData={cleanData} getApiQuery={getApiQuery} heroBanner={heroBanner} />
						<DataBanner showContent={showContent} queryValue={queryValue} cardsData={retrievedObjet} />
					</section>
				)}
			</section>
		)

	}
}

HomePage.propTypes = {
	retrievedObjet: PropTypes.arrayOf(PropTypes.any),
	tflInfo: PropTypes.arrayOf(PropTypes.any),
	getApiQuery: PropTypes.func,
	queryValue: PropTypes.string,
	showContent: PropTypes.bool,
	cleanData: PropTypes.func
}

HomePage.defaultProps = {
	tflInfo: undefined,
	retrievedObjet: undefined,
	queryValue: undefined,
	showContent: false
}

export default HomePage
