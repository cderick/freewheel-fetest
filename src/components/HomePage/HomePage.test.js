import React from 'react'
import renderer from 'react-test-renderer'
import HomePage from './HomePage'

describe('Test on <HomePage /> component', () => {
   test('should render with no props', () => {
      const componentNoprops = renderer.create(<HomePage />)
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
})