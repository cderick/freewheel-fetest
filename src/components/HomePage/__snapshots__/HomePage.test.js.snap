// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <HomePage /> component should render with no props 1`] = `
<section>
  <section>
    <nav
      className="topnav"
      id="home"
    >
      <a
        className="float-left"
        href="https://www.freewheel.com/"
        id="home"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="freewheel-logo"
          className="imgResize"
          src="https://www.freewheel.com/wp-content/themes/understrap-child/img/FREEWHEEL-KO.png"
        />
      </a>
      <button
        onClick={[Function]}
      >
        Hire a Bike
      </button>
      <button
        onClick={[Function]}
      >
        TFL Information
      </button>
    </nav>
  </section>
  <section>
    <section>
      <div
        className="container-fluid containerBackground"
      >
        <div
          className="row h-100"
        >
          <div
            className="col my-auto text-center"
          >
            <h1
              className="display-4"
            >
              TFL Information
            </h1>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-5 m-5 infoService"
        />
      </div>
    </section>
  </section>
</section>
`;
